javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,951 @@
+      val size = strbuf.length
+      if (count != size) {
+        return false
+      }
+      regionMatches(0, new _String(0, size, strbuf.getValue), 0, size)
+    }
+  }
+
+  def contentEquals(cs: CharSequence): scala.Boolean = {
+    val len = cs.length
+    if (len != count) {
+      return false
+    }
+    if (len == 0 && count == 0) {
+      return true
+    }
+    regionMatches(0, cs.toString, 0, len)
+  }
+
+  def matches(expr: _String): scala.Boolean = Pattern.matches(expr, this)
+
+  def replaceAll(expr: _String, substitute: _String): _String = {
+    Pattern.compile(expr).matcher(this).replaceAll(substitute)
+  }
+
+  def replaceFirst(expr: _String, substitute: _String): _String = {
+    Pattern.compile(expr).matcher(this).replaceFirst(substitute)
+  }
+
+  def split(expr: _String): Array[_String] = Pattern.compile(expr).split(this)
+
+  def split(expr: _String, max: Int): Array[_String] = Pattern.compile(expr).split(this, max)
+
+  def subSequence(start: Int, end: Int): CharSequence = sub_String(start, end)
+
+  def codePointAt(index: Int): Int = {
+    if (index < 0 || index >= count) {
+      throw new IndexOutOfBoundsException()
+    }
+    val s = index + offset
+    java.lang.Character.codePointAt(value, s, offset + count)
+  }
+
+  def codePointBefore(index: Int): Int = {
+    if (index < 1 || index > count) {
+      throw new IndexOutOfBoundsException()
+    }
+    val s = index + offset
+    java.lang.Character.codePointBefore(value, s)
+  }
+
+  def codePointCount(beginIndex: Int, endIndex: Int): Int = {
+    if (beginIndex < 0 || endIndex > count || beginIndex > endIndex) {
+      throw new IndexOutOfBoundsException()
+    }
+    val s = beginIndex + offset
+    java.lang.Character.codePointCount(value, s, endIndex - beginIndex)
+  }
+
+  def contains(cs: CharSequence): scala.Boolean = indexOf(cs.to_String) >= 0
+
+  def offsetByCodePoints(index: Int, codePointOffset: Int): Int = {
+    val s = index + offset
+    val r = java.lang.Character.offsetByCodePoints(value, offset, count, s, codePointOffset)
+    r - offset
+  }
 }
 
 object _String {
-  def valueOf(obj: Object): _String = ???
-  def valueOf(data: Array[Char]): _String = ???
-  def valueOf(data: Array[Char], offset: scala.Int, count: scala.Int): _String = ???
-  def valueOf(b: scala.Boolean): _String = ???
-  def valueOf(c: scala.Char): _String = ???
-  def valueOf(s: scala.Short): _String = ???
-  def valueOf(i: scala.Int): _String = ???
-  def valueOf(l: scala.Long): _String = ???
-  def valueOf(f: scala.Float): _String = ???
-  def valueOf(d: scala.Double): _String = ???
-  def format(format: String, args: scala.Array[Object]): _String = ???
+  object ConsolePrintStream {
+    private var charset: _String =
+      AccessController.doPrivileged(new PriviAction[_String]("console.encoding", "ISO8859_1"))
+
+    if (!Charset.isSupported(charset)) {
+      charset = "ISO-8859-1"
+    }
+  }
+
+  class ConsolePrintStream(out: java.io.OutputStream)
```

Comment:
`ConsolePrintStream` should be in a class, not in the companion. Otherwise it will be binary incompatible.